cmd/geranos/s3uploader.go

```diff
@@ -197,15 +199,19 @@ func keyForImageRecord(imageDigest string) string {
 }
 
 func (s *s3Uploader) blobExists(bucket, key string) (bool, error) {
-	_, err := s.svc.HeadObject(&s3.HeadObjectInput{
+	_, err := s.svc.HeadObject(context.TODO(), &s3.HeadObjectInput{
 		Bucket: aws.String(bucket),
 		Key:    aws.String(key),
 	})
 	if err != nil {
-		// yes, we really have to typecast to compare against an undocument string
-		// to check if the object doesn't exist vs an error making the call
```

Comment:
this comment is still true? "NotFound" is an undocumented error string, this code smells but it's the only way to implement this (I would ordinarily expect in a code review to be comparing to an exported error or constant)

Reply:
the `if errors.As(err, &notFound) {` should work! but i saw one report of it may be not working right, so i added the previous check as fallback